.codegen/service.go.tmpl

```diff
@@ -73,6 +73,23 @@ func New() *cobra.Command {
 		// This service is being previewed; hide from help output.
 		Hidden: true,
 		{{- end }}
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
If I understand correctly, `RunE` is only executed if the subcommand is not found.

Would it be possible to instead of specifying a custom `RunE`, to specify an `Args: cobra.NoArgs`, or custom function that prints something about the subcommand. I doubt this needs to resolve the subcommand.

Reply:
Unfortunately not, this is what I was trying to do in the first place. So the args validation happens in `cmd.execute` (https://github.com/spf13/cobra/blob/ceb39aba25c86233e4888210c4b57cdb0a78d1e1/command.go#L905) which is called only after c.Find call (which searches for subcommands) https://github.com/spf13/cobra/blob/ceb39aba25c86233e4888210c4b57cdb0a78d1e1/command.go#L1123-L1148
And Find calls this legacyArgs and returns an error before validate args are even executed https://github.com/spf13/cobra/blob/ceb39aba25c86233e4888210c4b57cdb0a78d1e1/command.go#L776

Reply:
Thanks. Very complicated.